common/physics-system.cpp

```diff
@@ -67,14 +67,22 @@ namespace tec {
 			if (!collidable) {
 				continue;
 			}
-			if (state.positions.find(entity_id) != state.positions.end()) {
-				glm::vec3 position = state.positions.at(entity_id).value;
+			// fill in the transform for our collidable from the current state
+			auto position_iter = state.positions.find(entity_id);
+			if (position_iter != state.positions.end()) {
+				glm::vec3 position = position_iter->second.value;
 				if (std::isfinite(position.x) && std::isfinite(position.y) && std::isfinite(position.z)) {
 					collidable->motion_state.transform.setOrigin(btVector3(position.x, position.y, position.z));
 				}
 			}
-			if (state.orientations.find(entity_id) != state.orientations.end()) {
-				glm::quat orientation = state.orientations.at(entity_id).value;
+			else {
+				// no position! that's not good
+				// wait to add the physics body to the world for now
+				continue;
+			}
```

Comment:
This is to try and work around some race condition between positions being set, it doesn't fix the race though